plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java

```diff
@@ -132,7 +132,11 @@ protected void registerTableFromResources(String table, String resourcePath, Que
                     .filter(resourceInfo -> resourceInfo.getResourceName().startsWith(resourcePath + "/"))
                     .collect(toImmutableList());
             for (ClassPath.ResourceInfo resourceInfo : resources) {
-                String fileName = resourceInfo.getResourceName().replaceFirst("^" + Pattern.quote(resourcePath), quoteReplacement(targetDirectory));
+                String fileName = resourceInfo.getResourceName()
+                        .replaceFirst("^" + Pattern.quote(resourcePath), quoteReplacement(targetDirectory))
+                        // Replace '%' (corresponding to '%' character url encoded) with '%25' in order
```

Comment:
Relates potentially to the false-negative found initially in https://github.com/trinodb/trino/issues/15376